src/logger.py

```diff
@@ -26,18 +30,26 @@ class Logger:
         self.logger = logger if logger else Logger(show_message=False).logger
 
     """
-    def __init__(self, log_level=logging.INFO, 
-                 logfile="logs/logs.txt",
+    def __init__(self,
+                 logger_name="default",
+                 log_level=logging.INFO, 
+                 logfile="logs/ols.logs",
                  show_message=False):
         msg=f"""
         ############################################################################
-        Set LOG_LEVEL environment variable (e.g., INFO, DEBUG) to cocntrol verbosity
+        Set LOG_LEVEL or LOG_LEVEL_CONSOLE environment variable (e.g., INFO, DEBUG) 
+        to control general logging verbosity or console specific logging level
         ############################################################################
         """
         if show_message: print(msg)
 
+        self.logger_name=logger_name
         self.log_level=os.getenv("LOG_LEVEL", log_level)
+        self.log_level_console=os.getenv("LOG_LEVEL_CONSOLE", self.log_level)
```

Comment:
does this need to explicitly invoke load_dotenv()? Otherwise i'm not sure you can be guaranteed load_dotenv() was called before you get here.  (you can't assume config was init'd before someone instantiates a logger, right?)

Reply:
In ols.py Config is called and that is the one that create the logger instance. So, that way we know dotenv was invoked before creating the logger instance

Reply:
seems unsafe to assume no future refactor is going to invalidate the assumption about that ordering.

At a minimum some sort of comment/documentation is needed about the assumptions this code is making/how it can be invoked.